src/webapi/html_collection.rs

```diff
@@ -0,0 +1,96 @@
+use webcore::value::{Value, Reference};
+use webcore::reference_type::ReferenceType;
+use webcore::try_from::TryInto;
+use webapi::element::Element;
+
+/// The HTMLCollection interface represents a generic collection
+/// (array-like object similar to arguments) of elements (in document order)
+/// and offers methods and properties for selecting from the list.
+/// 
+/// An HTMLCollection in the HTML DOM is live; it is automatically
+/// updated when the underlying document is changed.
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection)
+// https://dom.spec.whatwg.org/#interface-htmlcollection
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "HTMLCollection")]
+pub struct HtmlCollection( Reference );
+
+impl HtmlCollection {
+    /// Returns the number of elements in the collection.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection)
+    // https://dom.spec.whatwg.org/#dom-htmlcollection-length
+    pub fn len( &self ) -> u32 {
+        js!( return @{self}.length; ).try_into().unwrap()
+    }
+
+    /// Returns an element from an HtmlCollection by index.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection/item)
+    // https://dom.spec.whatwg.org/#dom-htmlcollection-item
+    pub fn item( &self, index: u32 ) -> Option< Element > {
+        js!(
+            return @{self}[ @{index} ];
```

Comment:
I'm not so sure about that: I know that [some browsers can optimize `[]` a *lot* better than `.item()`](https://jsperf.com/nodelist-item-x-vs-nodelist-x2). (e.g. Chrome runs `[]` over twice as fast!)

Of course [microbenchmarks should be taken with a *very* hefty grain of salt](https://www.youtube.com/watch?v=g0ek4vV7nEA), but still.

Reply:
NodeList (node_list.rs) also use `[]`, do you what to change it to `.item` as well?

Reply:
Run benchmark linked by @Pauan on my system: 

* Firefox 59: `[]` 41% slower than `.item`
* Chromium 65: `.item` 46% slower than `[]`

So, I will use `.item` for an `.item`.

Note that what I have done for `html_collection.rs` is copy from `node_list.rs`. After this discussion, I found out that I can use `HtmlCollection::item` in its iterator (in `next` method). Please take a look to `html_collection.rs` (`item`, `ElementIter`), are they ok? (they are covered by tests in `select.rs`)

Side note: While working on this I found out that `node_list.rs` use an `i32` in its iterator instead of `u32`.